storage/Devices/EncryptionImpl.h

```diff
@@ -56,6 +56,13 @@ namespace storage
 
 	virtual string get_displayname() const override { return get_dm_table_name(); }
 
+	/**
+	 * Redefined from BlkDeviceImpl to ensure the device name is in sync
+	 * with the DeviceMapper name, since the kernel device names for
+	 * Encryption devices are inferred from their DeviceMapper names.
+	 */
+	void set_dm_table_name(const string& dm_table_name);
```

Comment:
NP: I usually add `virtual` in the sub-classes too, to make it more explicit. If you only look at `EncryptionImpl` you don't know if `set_dm_table_name` is virtual or not. But only a matter of taste.

Reply:
Fixed. I'm a C++ noob.